src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -1438,7 +1491,16 @@ namespace cryptonote
               if (completion_percent == 100) // never show 100% if not actually up to date
                 completion_percent = 99;
               progress_message = " (" + std::to_string(completion_percent) + "%, "
-                  + std::to_string(target_blockchain_height - current_blockchain_height) + " left)";
+                  + std::to_string(target_blockchain_height - current_blockchain_height) + " left";
+              std::string time_message = get_periodic_sync_estimate(current_blockchain_height, target_blockchain_height);
+              if (!time_message.empty())
+              {
+                uint64_t total_blocks_to_sync = target_blockchain_height - m_sync_start_height;
+                uint64_t total_blocks_synced = current_blockchain_height - m_sync_start_height;
+                progress_message += ", " + std::to_string(total_blocks_synced * 100 / total_blocks_to_sync) + "% of total synced";
```

Comment:
Is total_blocks_to_sync guaranteed to not be 0 here ?

Reply:
Yes, together by the 2 `if` statements that guard that code and the way `m_start_sync_height` was initialized:

    if (current_blockchain_height > previous_height)

    if (current_blockchain_height < target_blockchain_height)

But maybe one of the cases where one should err on the safe side and add a check for zero nevertheless, for "peace of mind" and to guard against later changes that may shift conditions?